UI/org.eclipse.birt.report.designer.ui.preview.web/src/org/eclipse/birt/report/designer/ui/ide/explorer/AbstractViewHandler.java

```diff
@@ -0,0 +1,43 @@
+package org.eclipse.birt.report.designer.ui.ide.explorer;
+
+import org.eclipse.core.commands.AbstractHandler;
+import org.eclipse.core.expressions.IEvaluationContext;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.jface.viewers.IStructuredSelection;
+import org.eclipse.ui.ISources;
+
+/**
+ *
+ * Abstract handler used for all handlers in the explorer
+ *
+ */
+abstract class AbstractViewHandler extends AbstractHandler {
```

Comment:
Honestly, I do not understand why we need this class
1) we have "visibleWhen" expression that checks for file (it could check more if needed)
2) we can use `HandlerUtil#getCurrentStructuredSelection(ExecutionEvent)` 
So, I would remove this class

Reply:
I will remove the class.